planet/clients/orders.py

```diff
@@ -433,3 +458,48 @@ async def list_orders(self, state: str = None, limit: int = None):
 
         request = self._request(url, 'GET', params=params)
         return Orders(request, self._do_request, limit=limit)
+
+
+def validate_checksum(manifest_data: dict, filenames: list, checksum: str):
```

Comment:
This looks great! Since the user can call this function directly, as it is not a private function just being used by `download_order` anymore, I would consider usage scenarios and look at adding a bit of defensive logic. For example, I am curious what happens if `checksum` is an invalid entry like `checksum='foobar'`. Additionally, the user could call this function with arbitrary entries in `filenames`. I am curious what happens if one or more of the entries in `filenames` is not contained within `manifest_data`. Also what happens if an entry in `filenames` does not exist? These scenarios may be good candidates for raising an exception. Since this logic is no longer contained within a client, it may make sense to use a `PlanetError` instead of a `ClientError` (and catch and re-raise in `download_order` as a `ClientError`). 

It also may make sense to add tests for these scenarios. The tests can test `validate_checksum` directly.

Reply:
@mkshah605 My recommendation here is to separate these efforts, go back to `validate_checksum` being a function within the client (I would make it private at this point so only `download_order` uses it) and merge the work for checksum matching in download orders. The work for checksum matching in download_orders is ready to be merged now and is required for the Orders CLI milestone, which is tomorrow (Wed). The effort to elevate `validate_checksum` involves more work and is not high priority.

Reply:
Thank you @jreiberkyle sounds good! I've included this info for added detail/future reference in [this ticket](https://github.com/planetlabs/planet-client-python/issues/521)